### PR TITLE
Add paragraph spacing for blank lines in MarkdownToHTML

### PR DESCRIPTION
**Pull request by nnemirovsky**
## Summary

Basecamp's Trix editor does not render margins on `<p>` tags, so consecutive paragraphs produced by `MarkdownToHTML` appear visually merged with no spacing between them.

This made it impossible to create visible paragraph separation when writing Markdown content via the CLI — blank lines in the input were silently discarded.

### Changes

- Blank lines between content blocks now emit `<br>` for visible paragraph separation in Basecamp
- Multiple consecutive blank lines collapse to a single `<br>`, matching standard Markdown paragraph-break semantics
- Leading blank lines (before any content) are still ignored
- Works for all block transitions: paragraph→paragraph, heading→paragraph, paragraph→list, paragraph→code block

### Before

```
Input:  "First paragraph\n\nSecond paragraph"
Output: "<p>First paragraph</p>\n<p>Second paragraph</p>"
```
No visible spacing in Basecamp.

### After

```
Input:  "First paragraph\n\nSecond paragraph"
Output: "<p>First paragraph</p>\n<br>\n<p>Second paragraph</p>"
```
Visible paragraph separation in Basecamp.

---
## Summary by cubic
Add visible paragraph spacing in HTML output from `MarkdownToHTML` by inserting <br> for non-leading blank lines. Fixes merged paragraphs and block transitions in Basecamp’s Trix editor.

- **Bug Fixes**
  - Emit one <br> for non-leading blank lines; collapse multiples; ignore leading.
  - Apply before lists, code blocks, blockquotes, horizontal rules, and between headings and paragraphs.
  - Add tests for paragraphs, headings, lists, code blocks, blockquotes, hr, and edge cases.
  - Remove `<br><br>` workaround from CLI notes and `skills/basecamp/SKILL.md`; extract `flushPendingBreak()` helper.

<sup>Written for commit b267159ed178a70e88ae4eb2910496dc82504f08. Summary will update on new commits.</sup>

### CODE REVIEW

File: internal/richtext/richtext.go [hunk 1]
```diff
@@ -149,6 +150,10 @@ func MarkdownToHTML(md string) string {
 			} else {
 				// Start code block
 				flushList()
+				if pendingBreak {
+					result.WriteString("<br>\n")
+					pendingBreak = false
+				}
```
[the remark this answers is not in the document]
Comment (nnemirovsky) [reply internal/richtext/richtext.go]: done in b267159 — extracted a flushPendingBreak() helper that replaces all four duplicated blocks

File: internal/richtext/richtext_test.go [hunk 2]
```diff
@@ -116,6 +116,36 @@ func TestMarkdownToHTML(t *testing.T) {
 			input:    "Tom & Jerry",
 			expected: "<p>Tom &amp; Jerry</p>",
 		},
+		{
+			name:     "paragraph spacing with blank line",
+			input:    "First paragraph\n\nSecond paragraph",
+			expected: "<p>First paragraph</p>\n<br>\n<p>Second paragraph</p>",
+		},
```
[the remark this answers is not in the document]
Comment (nnemirovsky) [reply internal/richtext/richtext_test.go:124]: done in b267159 — added test cases for blank-line-before-blockquote and blank-line-before-hr
